src/main/java/com/playmonumenta/networkrelay/NetworkMessageListenerBungee.java

```diff
@@ -130,6 +125,22 @@ public void destOnline(DestOnlineEventBungee event) {
 			return;
 		}
 
+		// move duplicate server check after to ensure that addresses are the same
+		if (getServers().containsKey(name)) {
+			SocketAddress previousAddress = getServers().get(name).getSocketAddress();
+			// check if addresses are different
+			if (previousAddress != null && previousAddress != addr) {
```

Comment:
I'm not sure if a SocketAddress is a singleton or not - you may need `!previousAddress.equals(addr)` instead.